### PR TITLE
[Fuzzer] Add two new ORT libfuzzer (Linux clang support for now)

### PR DESCRIPTION
**Pull request by 0xdr3dd**
### Description
This PR adds two new libfuzzer in fuzzer project.
1. Binary libfuzzer 
2. libprotobuf-fuzzer

To compile run below cmd on linux:
```
LLVM_PROFILE_FILE="%p.profraw" CFLAGS="-g -fsanitize=address,fuzzer-no-link -shared-libasan -fprofile-instr-generate -fcoverage-mapping" CXXFLAGS="-g -shared-libasan -fsanitize=address,fuzzer-no-link -fprofile-instr-generate -fcoverage-mapping" CC=clang CXX=clang++ ./build.sh --update --build --config Debug --compile_no_warning_as_error --build_shared_lib --skip_submodule_sync --use_full_protobuf  --parallel --fuzz_testing --build_dir build/
```
Run fuzzer:
```
LD_PRELOAD=$(clang -print-file-name=libclang_rt.asan-x86_64.so) build/Debug/onnxruntime_libfuzzer_fuzz  testinput -rss_limit_mb=8196 -max_total_time=472800 -fork=2 -jobs=4 -workers=4 -ignore_crashes=1 -max_len=2097152 2>&1 | grep -v "\[libprotobuf ERROR"
```

### Motivation and Context
The existing custom fuzzer is not coverage guided and it's slow and it will work on one model mutation at a time. The new fuzzers are coverage guided, and we can use more models' files as a corpus to increase the coverage.

### CODE REVIEW

File: onnxruntime/test/fuzzing/ort_libfuzzer/OrtProtoLibfuzzer.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+      // Access the graph from the model
+      auto* graph = model_proto->mutable_graph();
+
+      // Set a random name for the graph
+      graph->set_name(generate_random_string(10, rng));
+    }};
+
+DEFINE_PROTO_FUZZER(const onnx::ModelProto& msg) {
+  Ort::Env env;
```
Comment (snnn) [inline onnxruntime/test/fuzzing/ort_libfuzzer/OrtProtoLibfuzzer.cpp:81]: Ort::Env should not be created/destroyed multiple times in a single process.  Will this function be called multiple times per process? Or, can we have a main function for this test process and manually init the env var there?

File: onnxruntime/test/fuzzing/ort_libfuzzer/OrtProtoLibfuzzer.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  try {
+    auto seed = static_cast<unsigned int>(std::chrono::system_clock::now().time_since_epoch().count());
+    onnx::ModelProto msg_proto = msg;
+    predict(msg_proto, seed, env);
+  } catch (const std::exception& e) {
+    // Optionally log or suppress the exception
+    // std::cerr << "Caught exception: " << e.what() << std::endl;
+  } catch (...) {
+    // Handle any other exceptions
```
Comment (snnn) [inline onnxruntime/test/fuzzing/ort_libfuzzer/OrtProtoLibfuzzer.cpp:90]: Why silently ignore all exceptions?

Comment (0xdr3dd) [reply onnxruntime/test/fuzzing/ort_libfuzzer/OrtProtoLibfuzzer.cpp:90]: there will be too many exceptions, if we will print those then fuzzer will be slowed down. basically, fuzzer only targets valid crashes not exceptions.

Comment (0xdr3dd) [reply onnxruntime/test/fuzzing/ort_libfuzzer/OrtProtoLibfuzzer.cpp:90]: even though ort will print those exceptions, so we don't need to handle it here.

File: onnxruntime/test/fuzzing/ort_libfuzzer/OrtProtoLibfuzzer.cpp (hunk 1 above)
Comment (0xdr3dd) [reply onnxruntime/test/fuzzing/ort_libfuzzer/OrtProtoLibfuzzer.cpp:81]: Yup, you are right this should be initialized only once. i will update this. Thanks.

Comment (0xdr3dd) [reply onnxruntime/test/fuzzing/ort_libfuzzer/OrtProtoLibfuzzer.cpp]: done

**Review by snnn — APPROVED** [review]

**Pull request merged**